### ISSUE TITLE
Halloween related assets don't render properly in KMS 1.2.370

### ISSUE THREAD

**Issue by HikariCalyx** [Kagamia/WzComparerR2#216]
Affected maps: Henesys, Leafre and Nameless Town.
![image](https://user-images.githubusercontent.com/29157608/201503469-a2479a40-3f10-458d-8ee5-78aeafd060bb.png)
![image](https://user-images.githubusercontent.com/29157608/201503498-5d90d761-84c3-451d-aaa7-d712b11fee1d.png)
![image](https://user-images.githubusercontent.com/29157608/201503482-0672f085-f6c0-4595-9ef2-72b73f6fc84a.png)

**Comment by Kagamia** [comment]
I don't have installed KMS client, also I can't see halloween elements in KMST 1147.
Is it only included in KMS?

**Comment by HikariCalyx** [comment]
Yes, Halloween assets weren't present in KMST 1.2.1147, only included in KMS.

**Comment by Kagamia** [comment]
> https://youtu.be/2xZLvReSmZk?t=9707

Cool. Many thanks.

**Comment by nanochromatic** [comment]
KMS 1.2.372 introduced Christmas themed assets which behave similarly.

![image](https://user-images.githubusercontent.com/5650392/209081004-c560c859-3f61-46a0-afbd-76a3e52fe184.png)

**Comment by HikariCalyx** [comment]
https://youtu.be/TSqFZ7mNYmw?t=65
in-game behavior as reference

**Comment by Kagamia** [comment]
Thanks for acknowledgement, really helps.

As it is the end of year, the deployment progress may be delayed for a long time. We don't have an ETA for the fix. (Believe me it is on the way)

**Comment by HikariCalyx** [comment]
After I use updated HaCreator I think I know what were those black tiles.
![image](https://user-images.githubusercontent.com/29157608/212551713-a290c986-0d13-41f9-80e9-fbbce3948500.png)

White part on these black tiles were used for glowing effect.

**Comment by Kagamia** [comment]
![image](https://user-images.githubusercontent.com/10445904/215782688-93a21cdc-9930-4e13-9fad-3ca72ff87a73.png)
the recent commit fixes the glowing object.

Still want some more clear screenshot (not only capture from videos) for global light rendering.
I got into trouble on global lighting simulation.

**Comment by HikariCalyx** [comment]
Here you go. Didn't find any ss regarding to Leafre yet.
![1](https://cdn.discordapp.com/attachments/344103015730642945/1035030267355738204/unknown.png)
![2](https://cdn.discordapp.com/attachments/344103015730642945/1035030472629157989/unknown.png)

Also the BGM mentioned in <MapCode>.img\info\bgmSub\<randomCode>\uol will be played in game instead.

**Comment by HikariCalyx** [comment]
Just a follow-up with this issue:
Now glowing items doesn't look right.
![image](https://github.com/Kagamia/WzComparerR2/assets/29157608/4ef5baff-0689-43e6-944d-2750011f82a5)

For your convenience, here's the 450001000.img with Halloween assets - load it along with CMS v201 client.

[450001000_halloween.zip](https://github.com/Kagamia/WzComparerR2/files/12721992/450001000_halloween.zip)

**Comment by Kagamia** [comment]
@HikariCalyx  ack.
Not sure if it is a regression, will take a look this week.

**Comment by Kagamia** [comment]
Can you take a screenshot in game?  
Someone tell me that **CMS 201 has not enabled halloween event**, curious where you get the client file.

**Comment by HikariCalyx** [comment]
> Can you take a screenshot in game? Someone tell me that **CMS 201 has not enabled halloween event**, curious where you get the client file.


Open CMS v201 client, then open these 3 img files individually.
After that, click 450001000.img inside of list, and click MapRenderer.

**Comment by Kagamia** [comment]
The operation is **meaningless**. 

A merged client is not a **Complete** client, it is equivalent to a **self-modified version** or a private server, which is not considered by wcR2's scope of support.

**Comment by nanochromatic** [comment]
@Kagamia The Halloween themed maps are now present in MSEA v226.
- 100000000: Henesys
- 240000000: Leafre
- 450001000: Nameless Town

**Comment by PirateIzzy** [comment]
![20231003230610927_450001000](https://github.com/Kagamia/WzComparerR2/assets/34289651/84d82c43-49aa-4beb-953d-133f3a0df026)
Following up on what @nanochromatic posted above. This is a screenshot from 450001000. Malgeum showcased it in this video as well:
https://youtu.be/tOV2ID4ZIaQ?t=74

**Comment by Kagamia** [comment]
Thanks for acknowledgement.
There must be something different between KMS and MSEA client.
Fortunately I have an old KMS client version so I'll try to find it out.

I believe that it's time to implement a REAL global lighting system now, but before I start, mapRender needs a refactor to make it easy to filter components with condition.

There's no ETA for the feature, but I can share a POC demo when it is ready.

**Comment by PirateIzzy** [comment]
![image (2)](https://github.com/Kagamia/WzComparerR2/assets/34289651/32046368-e44b-4e05-b90b-5f891e6ba5ba)
Just as another follow-up, someone shared a screenshot of the MSEA version in-game.

**Comment by Kagamia** [comment]
![image](https://github.com/Kagamia/WzComparerR2/assets/10445904/cf447861-3d60-49c1-87a8-24af2047f7d5)
lol

the mimimap is totally wrong.

**Comment by Kagamia** [comment]
Find it.

obj | client-ver | z | canvas
---|---|---|---
450001000.img\3\obj\17 | KMS | 8 | ![Map/Obj/extinction.img/extinction/foothold0/6/0](https://github.com/Kagamia/WzComparerR2/assets/10445904/bdae1c05-6999-425a-9bad-766bddd12e46)
450001000.img\3\obj\17 | MSEA | 7 **<- here is the difference** | (same as above)
450001000.img\3\obj\54 | KMS | 7 | ![Map/Obj/halloween_2016.img/noName/obj_ani/30/0](https://github.com/Kagamia/WzComparerR2/assets/10445904/53680298-a681-4792-9867-6f76d9c98bca)
450001000.img\3\obj\54 | MSEA | 7 | (same as above)

The glowing object should have lower z than the lit object, but in MSEA it upside down.

WcR2 does not really support global light yet, just render the light obj with *additive blend* instead of *alpha blend* to make it looks better.

Despite all this, It seems to prove that global lighting ignores z-layer sorting, it is a good news that we still have new discoveries from the buggy client.

**Comment by Kagamia** [comment]
progress: 40%

![image](https://github.com/Kagamia/WzComparerR2/assets/10445904/385b3948-8b79-4e6f-936b-bfbd8cc48891)

**Comment by PirateIzzy** [comment]
Just wanted to provide an update from KMST 1.2.162, these are the latest map renders:

![20231019211030791_100000000](https://github.com/Kagamia/WzComparerR2/assets/34289651/511a4c90-3bab-4c9e-ba71-604a794e6c7e)
![20231019211009694_450001000](https://github.com/Kagamia/WzComparerR2/assets/34289651/3f1fe094-3c31-41d9-bff5-26b8cd12ae1a)
![20231019211020574_410000500](https://github.com/Kagamia/WzComparerR2/assets/34289651/9c6b958a-d78f-4bb5-aa69-9ad402db1c5a)

None for Leafre

**Comment by Kagamia** [comment]
Looks some light objects are not tracked. will take a look tomorrow.

**Comment by Kagamia** [comment]
![image](https://github.com/Kagamia/WzComparerR2/assets/10445904/e7e94ef3-4d19-49d6-be5d-0d70af76f584)

🤣
